### PR TITLE
To cut a dependency hell for now, we'll copy some code

### PR DESCRIPTION
**Pull request by pimeys**
THIS IS WHAT IT ALL MEANS PEOPLE!

Copied the `ToSql`/`FromSql` code from rust-decimal to cut a bad dependency hell and get many of our deps back to crates.io. Some of the code is cleaned using the knowledge from SQLx, but it's still quite a lot :(

### CODE REVIEW

File: src/connector/postgres/conversion/decimal.rs [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+    fn accepts(ty: &Type) -> bool {
+        match ty {
+            &Type::NUMERIC => true,
+            _ => false,
+        }
+    }
+
+    to_sql_checked!();
+}
```
Comment (tomhoule) [inline src/connector/postgres/conversion/decimal.rs:293]: Are there tests we could also copy?

**Review by tomhoule — APPROVED** [review]
ok ok ok

File: src/connector/postgres/conversion/decimal.rs (hunk 1 above)
Comment (pimeys) [reply src/connector/postgres/conversion/decimal.rs:293]: I don't know do I want those tests...

Comment (pimeys) [reply src/connector/postgres/conversion/decimal.rs:293]: Need to write my own.

Comment (pimeys) [reply src/connector/postgres/conversion/decimal.rs:293]: I'll do that before merging.

**Pull request merged**